src/ReservoirPriceOracle.sol

```diff
@@ -545,6 +556,9 @@ contract ReservoirPriceOracle is IPriceOracle, IReservoirPriceOracle, Owned(msg.
             address lThirdToken = aRoute[2];
 
             if (lRouteLength == 3) {
+                // REVIEW: I'm not sure how these 2 or 3 hop routes work? As far
+                // as I can tell only a single address gets written to storage?
+                // What am I missing?
```

Comment:
for a length 3 routes, we only need to write one address, which is the one in the middle 

for length 4 routes, we only need to write two addresses in the middle. 

https://miro.com/app/board/uXjVKR-IxOM=/?moveToWidget=3458764593461460934&cot=14

Reply:
one improvement we should make for the length 4 routes is to store the second token wholly on the 2nd word instead of splitting it up

Reply:
implemented